tasmota/xsns_96_qmc5883.ino

```diff
@@ -0,0 +1,217 @@
+/*
+  xsns_96_qmc5883.ino - QMC5883 3-Axis Digital Compass sensor support for Tasmota
+
+  Copyright (C) 2022 James Sharp
+  
+  Code borrowed heavily from Elnur Mehdiyev's implementation for the HMC5883L
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_QMC5883
+
+/*********************************************************************************************\
+ * QMC5883 is 3-Axis Digital Compass sensor knock off of the HMC5883L
+ *
+ * Source: James Sharp, code heavily borrowed from Elnur Mehdiyev's project
+ *
+ * I2C Address: 0x1D
+\*********************************************************************************************/
+
+// Define driver ID
+#define XSNS_96                          96
+#define XI2C_66                          66  // See I2CDEVICES.md
+
+#define QMC5883_ADDR                    0x0D
+
+#define Mode_Standby    0b00000000
+#define Mode_Continuous 0b00000001
+
+#define ODR_10Hz        0b00000000
+#define ODR_50Hz        0b00000100
+#define ODR_100Hz       0b00001000
+#define ODR_200Hz       0b00001100
+
+#define RNG_2G          0b00000000
+#define RNG_8G          0b00010000
+
+#define OSR_512         0b00000000
+#define OSR_256         0b01000000
+#define OSR_128         0b10000000
+#define OSR_64          0b11000000
+
+
+//Mag field
+struct
+{
+int   MX;
+int   MY;
+int   MZ;
+int   HG;
+uint8_t   status;
+bool      ready;
+uint8_t   i2c_address;
+
+} QMC5883;

```

Comment:
Do not allocate statically the memory for your driver
This is taking useless amount of RAM when the driver is part of a build but not used

Use dynamic allocation as in the HMC5883 driver